modules/top-level/plugins/byte-compile-plugins.nix

```diff
@@ -4,15 +4,25 @@
   Inputs: List of normalized plugins
   Outputs: List of normalized (compiled) plugins
 */
-{ lib, pkgs }:
+{
+  lib,
+  pkgs,
+  uncompiledPlugins,
+}:
+normalizedPlugins:
 let
   builders = lib.nixvim.builders.withPkgs pkgs;
   inherit (import ./utils.nix lib) mapNormalizedPlugins;
 
+  uncompiledNames = map (p: if builtins.isString p then p else lib.getName p) uncompiledPlugins;
+  isUncompiled = p: builtins.elem (lib.getName p.plugin) uncompiledNames;
```

Comment:
Again, same nitpick. This isn't about whether it _is_ uncompiled, but whether it _should_ be.

I'm also cautious of negative boolean/predicate names. Things usually flow better and make more sense with positive names, though there's no hard-and-fast rule here.

Also, one entire unrelated nitpick: it may be necessary in some cases to compare by equality instead of name. E.g. we could take the plugin's `outPath` (its in-store path) or even directly use `==` on plugins.

How about:

```
shouldCompile =
  plugin:
  let
    name = lib.getName plugin;
  in
  lib.none (
    excluded:
    if lib.isString excluded then
      name == excluded
    else
      plugin == excluded
      # or: plugin.outPath == excluded.outPath
  ) excludedPlugins;
```

?

Reply:
> Maybe we should consider improving both in another PR then.

I want to point out that I changed it to `getName` in #3421. Literal comparison should work in this PR (if you want) because byte-compilation is the first step, but it won't work in combinePlugins (because byte-compilation is changing derivations). After I changed it to name comparison there was no point to compare it also with `==`, because if it's the same derivation, name would be equal anyway.

Reply:
Ah, I see. In that case #3421 shouldn't be reverted unless someone want's to explore one of the alternative solutions discussed there.

As for here, I could be convinced either way: on the one hand, having the ability to compare with `==` would be more powerful in very niche situations; on the other hand, having consistency would be more intuitive and less confusing.